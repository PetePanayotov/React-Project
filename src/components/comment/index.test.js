import React from 'react';
import {create , act} from 'react-test-renderer';
import UserContext from '../../Context';
import Comment from './index';


describe('Comment Component' , () => {

    it('Should render admin account comment' , () => {
        let root;
        act(() => {
            
            root = create(
                <UserContext.Provider value={{
                    isLoggedIn: true,
                    isAdmin: true,
                    user: {
                        username: 'tigyra',
                        userId: '1234'
                    }
                }}>

                    <Comment author="amador" comment="Vyrvi li" time="12.08.2020" handler={() => {}}/>
                </UserContext.Provider>
            );
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('Should render regular user account comment' , () => {
        let root;
        act(() => {
            
            root = create(
                <UserContext.Provider value={{
                    isLoggedIn: true,
                    isAdmin: false,
                    user: {
                        username: 'tigyra',
                        userId: '1234'
                    }
                }}>

                    <Comment author="amador" comment="Vyrvi li" time="12.08.2020" handler={() => {}}/>
                </UserContext.Provider>
            );
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

});