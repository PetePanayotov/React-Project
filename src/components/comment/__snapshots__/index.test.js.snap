// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component Should render admin account comment 1`] = `
<div
  className="container"
>
  <p
    className="author-paragraph"
  >
    amador
    <label
      className="label"
    >
      12.08.2020
    </label>
  </p>
  <p
    className="comment-paragraph"
  >
    Vyrvi li
  </p>
  <div
    className="remove-button-div"
  >
    <button
      className="remove-button"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
</div>
`;

exports[`Comment Component Should render regular user account comment 1`] = `
<div
  className="container"
>
  <p
    className="author-paragraph"
  >
    amador
    <label
      className="label"
    >
      12.08.2020
    </label>
  </p>
  <p
    className="comment-paragraph"
  >
    Vyrvi li
  </p>
</div>
`;
